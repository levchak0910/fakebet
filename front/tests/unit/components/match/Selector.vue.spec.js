import {createWrapper, disableFile} from "../../__utils__";
import Component from "@/components/match/Actions/Selector";

disableFile();

describe("match/Selector.vue", () => {

    let wrapper, sampleHeadSpan, noneMatches;
    let mountWrapper = (options = {}) => {
        wrapper = createWrapper(Component, {
            stubs: ["app-match-selector-item"],
            ...options,
        });
        sampleHeadSpan = wrapper.find("p.head > span:nth-of-type(1)");
        noneMatches = wrapper.find("#none-matches");
    };

    describe("Testing snapshots", () => {

        it("None matches element", () => {
            mountWrapper({computed: {emptySelectedMatches: () => true}});

            expect(noneMatches.element).toMatchSnapshot();
        });

        it("Match-selector", () => {
            mountWrapper();

            expect(wrapper.element).toMatchSnapshot();
        });

    });

    describe("Triggering event", () => {

        it("Triggering after click on head span", () => {
            mountWrapper({methods: ["sortMatches"]});

            sampleHeadSpan.trigger("click");

            expect(wrapper.vm.sortMatches).toBeCalled();
        });

    });

    describe("Testing computed properties", () => {

        it("emptySelectedMatches", () => {
            mountWrapper({computed: {matchesFromSelectedLeagues: () => []}});

            expect(wrapper.vm.emptySelectedMatches).toBeTruthy();
        });

        describe("matchesFromSelectedLeagues", () => {
            let makeIt = (type, direction, expected) => {
                it(`type=${type}, direction=${direction}`, () => {
                    mountWrapper({data: {sort: {type, direction}}});

                    expect(wrapper.vm.matchesFromSelectedLeagues[0].league).toBe(expected);
                });
            };

            makeIt("0", 1, "League3"); makeIt("0", -1, "League1");
            makeIt("1", 1, "League2"); makeIt("1", -1, "League3");
            makeIt("2", 1, "League1"); makeIt("2", -1, "League2");
            makeIt("Date", 1, "League3"); makeIt("Date", -1, "League1");
            makeIt("Teams", 1, "League3"); makeIt("Teams", -1, "League1");
        });

    });

    describe("Testing methods", () => {

        it("sortMatches", () => {
            mountWrapper();
            let customSort = {name: "sortByName", direction: 11};

            wrapper.vm.sortMatches(customSort);

            expect(wrapper.vm.sort.direction).toBe(11);
            expect(wrapper.vm.sort.type).toBe("sortByName");
            expect(customSort.direction).toBe(-11);
        });

    });

});
