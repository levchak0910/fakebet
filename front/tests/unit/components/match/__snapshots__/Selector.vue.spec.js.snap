// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match/Selector.vue Testing snapshots Match-selector 1`] = `
<main>
  <p
    class="row head"
  >
    <span
      class="date"
      title="Date. click.for.sort"
    >
      Date
    </span>
    <span
      class="teams"
      title="Teams. click.for.sort"
    >
      Teams
    </span>
    <span
      class="h"
      title="win.home. click.for.sort"
    >
      1
    </span>
    <span
      class="d"
      title="Draw. click.for.sort"
    >
      0
    </span>
    <span
      class="g"
      title="win.guest. click.for.sort"
    >
      2
    </span>
  </p>
  <p
    class="row"
  >
    <span
      class="date"
    >
      27.10
    </span>
    <span
      class="teams"
    >
      Home team 1 — Guest team 1
    </span>
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League1:Home team 1-Guest team 1"
      type="0"
    />
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League1:Home team 1-Guest team 1"
      type="1"
    />
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League1:Home team 1-Guest team 1"
      type="2"
    />
  </p>
  <p
    class="row"
  >
    <span
      class="date"
    >
      28.10
    </span>
    <span
      class="teams"
    >
      Home team 2 — Guest team 2
    </span>
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League2:Home team 2-Guest team 2"
      type="0"
    />
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League2:Home team 2-Guest team 2"
      type="1"
    />
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League2:Home team 2-Guest team 2"
      type="2"
    />
  </p>
  <p
    class="row"
  >
    <span
      class="date"
    >
      29.10
    </span>
    <span
      class="teams"
    >
      Home team 3 — Guest team 3
    </span>
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League3:Home team 3-Guest team 3"
      type="0"
    />
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League3:Home team 3-Guest team 3"
      type="1"
    />
    <app-match-selector-item-stub
      coefficients="[object Object],[object Object],[object Object]"
      matchkey="League3:Home team 3-Guest team 3"
      type="2"
    />
  </p>
</main>
`;

exports[`match/Selector.vue Testing snapshots None matches element 1`] = `
<p
  id="none-matches"
>
  not.found.matches
</p>
`;
